pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go

```diff
@@ -490,7 +490,12 @@ func (r *OpsGenieConfigResponder) Validate() error {
 // HTTPConfig defines a client HTTP configuration.
 // See https://prometheus.io/docs/alerting/latest/configuration/#http_config
 type HTTPConfig struct {
+	// Authorization header configuration for the client.
+	// This is mutually exclusive with BasicAuth and is only available starting from Alertmanager v0.22+.
+	// +optional
+	Authorization *monitoringv1.Authorization `json:"authorization,omitempty"`
```

Comment:
I'd stick with `SafeAuthorization`

```suggestion
	Authorization *monitoringv1.SafeAuthorization `json:"authorization,omitempty"`
```